src/web/services/student.service.ts

```diff
@@ -129,10 +132,34 @@ export class StudentService {
    * Loads list of students from a course in CSV format by calling API.
    */
   loadStudentListAsCsv(queryParams: { courseId: string }): Observable<string> {
-    const paramsMap: Record<string, string> = {
+    return this.httpRequestService.get(ResourceEndpoints.COURSE, {
       courseid: queryParams.courseId,
-    };
-    const responseType: string = 'text';
-    return this.httpRequestService.get(ResourceEndpoints.STUDENTS_CSV, paramsMap, responseType);
+      entitytype: 'instructor',
+    }).pipe(mergeMap((course: Course) => {
+      return this.httpRequestService.get(ResourceEndpoints.STUDENTS, {
+        courseid: queryParams.courseId,
+      }).pipe(map((students: Students) => {
+        return this.processStudentsToCsv(course.courseId, course.courseName, students.students);
+      }));
+    }));
+  }
+
+  processStudentsToCsv(courseId: string, courseName: string, students: Student[]): string {
+    const csvRows: string[][] = [];
+    csvRows.push(['Course ID:', StringHelper.removeExtraSpace(courseId)]);
+    csvRows.push(['Course Name:', StringHelper.removeExtraSpace(courseName)]);
+    csvRows.push([]);
+    csvRows.push(['Section', 'Team', 'Full Name', 'Last Name', 'Status', 'Email']);
+    students.forEach((student: Student) => {
+      csvRows.push([
+        student.sectionName ? StringHelper.removeExtraSpace(student.sectionName) : '',
```

Comment:
I guess in the original Java code, we don't have `removeExtraSpace` for `sectionName`? Same for other fields.

Reply:
I went to double check and it was only used for fullNames & lastName, should I add `removeExtraSpace` in the frontend for them as well or just leave it as it is right now (without the method).

Reply:
I actually don't understand the reason behind removing extra space for the two specific fields (actually I think there is no need to do the operation). But for safe, let's just follow the Java code.

Reply:
This should hardly matter, because when it goes to the DB the section name, team name, and student name (and email) have all been trimmed.
But the problem here is those two are treated differently where the correct case should be everything treated the same; either remove the trailing/leading whitespace for all or for none.

Reply:
Okay, I think based on this discussion I'd rather head in the direction of not removing extra space for all. Even if there is an extra space it shouldn't impact the csv format, and having that extra overhead processing for every student row is going to impact performance on a large course since the generation is now being done client side instead of server side.